File: src/main/java/no/fint/provider/springer/model/SkoleRepository.java
package no.fint.provider.springer.model;

import lombok.extern.slf4j.Slf4j;
import no.fint.event.model.Event;
import no.novari.fint.model.resource.FintLinks;
import no.novari.fint.model.resource.utdanning.utdanningsprogram.SkoleResource;
import no.novari.fint.model.utdanning.utdanningsprogram.UtdanningsprogramActions;
import no.fint.provider.springer.storage.SpringerRepository;
import org.springframework.stereotype.Repository;

import java.util.Set;
import java.util.stream.Collectors;
import java.util.stream.Stream;

@Slf4j
@Repository
public class SkoleRepository extends SpringerRepository {

    @Override
    public void accept(Event<FintLinks> response) {
        if (UtdanningsprogramActions.valueOf(response.getAction()) == UtdanningsprogramActions.GET_ALL_SKOLE) {
            query(SkoleResource.class, response);
        }
    }

    @Override
    public Set<String> actions() {
        return Stream
            .of(UtdanningsprogramActions.GET_ALL_SKOLE)
            .map(Enum::name)
            .collect(Collectors.toSet());
    }
}

File: src/main/kotlin/no/fint/provider/springer/model/SkoleRepository.kt
package no.fint.provider.springer.model

import no.fint.event.model.Event
import no.fint.event.model.ResponseStatus
import no.fint.event.model.Status
import no.fint.provider.springer.storage.SpringerRepository
import no.fint.provider.springer.storage.Wrapper
import no.novari.fint.model.resource.FintLinks
import no.novari.fint.model.resource.utdanning.utdanningsprogram.ArstrinnResource
import no.novari.fint.model.resource.utdanning.utdanningsprogram.SkoleResource
import no.novari.fint.model.utdanning.utdanningsprogram.UtdanningsprogramActions
import org.springframework.data.mongodb.core.MongoTemplate
import org.springframework.stereotype.Repository

@Repository
class UtdanningsprogramRepository(
    mongoTemplate: MongoTemplate,
    wrapper: Wrapper
) : SpringerRepository(mongoTemplate, wrapper) {

    override fun accept(response: Event<FintLinks>) {
        when (UtdanningsprogramActions.valueOf(response.action)) {
            UtdanningsprogramActions.GET_ALL_SKOLE -> query(SkoleResource::class.java, response)
            UtdanningsprogramActions.GET_ALL_ARSTRINN -> query(ArstrinnResource::class.java, response)
            else -> {
                response.status = Status.ADAPTER_REJECTED
                response.responseStatus = ResponseStatus.REJECTED
                response.statusCode = "INVALID_ACTION"
                response.message = "Invalid action"
            }
        }
    }

    override fun actions(): Set<String> =
    setOf(
        UtdanningsprogramActions.GET_ALL_SKOLE,
        UtdanningsprogramActions.GET_ALL_ARSTRINN
    ).map { it.name }
     .toSet()
}
